diff --git a/src/schema.rs b/src/schema.rs
index 34f6b1a0..9c76fb1f 100644
--- a/src/schema.rs
+++ b/src/schema.rs
@@ -102,17 +102,31 @@ table! {
         id -> Int4,
         name -> Text,
         user_id -> Int4,
     }
 }

+table! {
+    recent_purchases(wine_id) {
+        wine_id -> Int4,
+        price -> Nullable<Float8>,
+        quantity -> Int4,
+        vintage -> Nullable<Int4>,
+        memo -> Nullable<Text>,
+        store_id -> Nullable<Int4>,
+        date -> Nullable<Date>,
+    }
+}
+
 joinable!(grapes -> users (user_id));
 joinable!(producers -> regions (region_id));
 joinable!(producers -> users (user_id));
 joinable!(purchases -> stores (store_id));
 joinable!(purchases -> wines (wine_id));
+joinable!(recent_purchases -> stores (store_id));
+joinable!(recent_purchases -> wines (wine_id));
 joinable!(stores -> users (user_id));
 joinable!(viti_areas -> regions (region_id));
 joinable!(viti_areas -> users (user_id));
 joinable!(wine_grapes -> grapes (grape_id));
 joinable!(wine_grapes -> wines (wine_id));
 joinable!(wine_types -> users (user_id));
@@ -131,7 +145,8 @@ allow_tables_to_appear_in_same_query!(
     stores,
     users,
     viti_areas,
     wine_grapes,
     wines,
     wine_types,
+    recent_purchases,
 );
